llama-index-core/llama_index/core/node_parser/interface.py

```diff
@@ -99,11 +99,17 @@ def get_nodes_from_documents(
                         )
 
                 if self.include_prev_next_rel:
-                    if i > 0:
+                    if (
+                        i > 0
+                        and nodes[i - 1].source_node.node_id == node.source_node.node_id
```

Comment:
I think we should also check if source_node is none or not too actually (I can imagine it being none sometimes)

Reply:
Good callout. Just pushed the change.